browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -207,6 +210,9 @@ BraveNewTabMessageHandler* BraveNewTabMessageHandler::Create(
 
 BraveNewTabMessageHandler::BraveNewTabMessageHandler(Profile* profile)
     : profile_(profile), weak_ptr_factory_(this) {
+  if (brave::UseAlternativeSearchEngineProviderEnabled(profile)) {
```

Comment:
We will show new toggle button again even if user turned off by using new toggle button?

Reply:
The idea here is that we'll only show the alternative search provider toggle button to the user if they've already enabled it in the past; if they never enabled the alternative search engine (which would be most users), we will no longer show the toggle that enables it on this page.

cc: @rebron to double-check my understanding of how this should work...